File: types/market.ts
export interface MarketMetric {
  symbol: string;
  name: string;
  price: number;
  change: number;
  changePercent: number;
  volume?: number;
  sparkline?: number[];
}

export interface MacroIndicator {
  name: string;
  value: number;
  unit: string;
  trend: "rising" | "falling" | "stable";
}

export interface SignalItem {
  ticker: string;
  name: string;
  signal: "bullish" | "bearish" | "neutral";
  confidence: number;
  expectedReturn: number;
}

export interface MarketOverviewResponse {
  indices: MarketMetric[];
  signals: SignalItem[];
  macro: MacroIndicator[];
  watchlist: WatchlistItem[];
  recentReports: ReportSummary[];
}

export interface WatchlistItem {
  ticker: string;
  name: string;
  price: number;
  change1d: number;
  change5d: number;
  change1m: number;
  signal: "bullish" | "bearish" | "neutral";
  signalScore: number;
  confidence: number;
  riskScore: number;
  alert?: string;
}

export interface ReportSummary {
  ticker: string;
  name: string;
  rating: string;
  confidence: number;
  generatedAt: string;
  summary: string;
}

File: backend/app/schemas/market.py
from __future__ import annotations

from pydantic import BaseModel
from typing import Optional


class MarketMetric(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    changePercent: float
    volume: Optional[int] = None
    sparkline: Optional[list[float]] = None


class MacroIndicator(BaseModel):
    name: str
    value: float
    unit: str
    trend: str  # "rising" | "falling" | "stable"


class SignalItem(BaseModel):
    ticker: str
    name: str
    signal: str
    confidence: float
    expectedReturn: float


class WatchlistItemBrief(BaseModel):
    ticker: str
    name: str
    price: float
    change1d: float
    change5d: float
    change1m: float
    signal: str
    signalScore: float
    confidence: float
    riskScore: float
    alert: Optional[str] = None


class ReportSummary(BaseModel):
    ticker: str
    name: str
    rating: str
    confidence: float
    generatedAt: str
    summary: str


class MarketOverviewResponse(BaseModel):
    indices: list[MarketMetric]
    signals: list[SignalItem]
    macro: list[MacroIndicator]
    watchlist: list[WatchlistItemBrief]
    recentReports: list[ReportSummary]
